flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BoundedBlockingSubpartitionReader.java

```diff
@@ -135,6 +135,11 @@ public Throwable getFailureCause() {
 		return null;
 	}
 
+	@Override
+	public int unsafeGetSizeOfQueuedBuffer() {
+		return parent.unsynchronizedGetNumberOfQueuedBuffers();
```

Comment:
I think this currently always returns `0`. Could you add a description to `flink/docs/monitoring/metrics.md` that `inputQueueLength`:

> ignores `LocalInputChannels` which are using blocking subpartitions.

I have also created a ticket for this: https://issues.apache.org/jira/browse/FLINK-12674